diff --git a/etherlink/lib_wasm_runtime/dune b/etherlink/lib_wasm_runtime/dune
index c5f0c0f2e3..577b009d4a 100644
--- a/etherlink/lib_wasm_runtime/dune
+++ b/etherlink/lib_wasm_runtime/dune
@@ -28,4 +28,4 @@
   (source_tree ../../src/kernel_sdk)
   (source_tree ../../sdk/rust)
   (source_tree src))
- (action (no-infer (bash ./build.sh))))
+ (action (no-infer (system "bash ./build.sh"))))
diff --git a/manifest/product_etherlink.ml b/manifest/product_etherlink.ml
index efc486e417..a66ffb8b37 100644
--- a/manifest/product_etherlink.ml
+++ b/manifest/product_etherlink.ml
@@ -103,7 +103,7 @@ let lib_etherlink_wasm_runtime =
                  [S "source_tree"; S "../../sdk/rust"];
                  [S "source_tree"; S "src"];
                ];
-               [S "action"; [S "no-infer"; [S "bash"; S "./build.sh"]]];
+               [S "action"; [S "no-infer"; [S "system"; S "bash ./build.sh"]]];
              ];
            ])
 
diff --git a/manifest/product_octez.ml b/manifest/product_octez.ml
index 3abf7623fe..57900aa2d9 100644
--- a/manifest/product_octez.ml
+++ b/manifest/product_octez.ml
@@ -534,7 +534,7 @@ let lib_rustzcash =
                  [S "source_tree"; S "include"];
                  [S "source_tree"; S "src"];
                ];
-               [S "action"; [S "no-infer"; [S "bash"; S "./build.sh"]]];
+               [S "action"; [S "no-infer"; [S "system"; S "bash ./build.sh"]]];
              ];
            ])
 
@@ -577,7 +577,7 @@ let octez_rust_deps, lib_wasmer_riscv =
                 [S "source_tree"; S "../kernel_sdk"];
                 [S "source_tree"; S "../../sdk/rust"];
               ];
-              [S "action"; [S "no-infer"; [S "bash"; S "./build.sh"]]];
+              [S "action"; [S "no-infer"; [S "system"; S "bash ./build.sh"]]];
             ];
           ]
   in
diff --git a/src/rust_deps/dune b/src/rust_deps/dune
index d724ca461c..dc07fd071f 100644
--- a/src/rust_deps/dune
+++ b/src/rust_deps/dune
@@ -24,4 +24,4 @@
   (source_tree ../riscv)
   (source_tree ../kernel_sdk)
   (source_tree ../../sdk/rust))
- (action (no-infer (bash ./build.sh))))
+ (action (no-infer (system "bash ./build.sh"))))
diff --git a/src/rustzcash_deps/dune b/src/rustzcash_deps/dune
index 9c60b89c8e..1216e9782b 100644
--- a/src/rustzcash_deps/dune
+++ b/src/rustzcash_deps/dune
@@ -20,4 +20,4 @@
   (source_tree bin)
   (source_tree include)
   (source_tree src))
- (action (no-infer (bash ./build.sh))))
+ (action (no-infer (system "bash ./build.sh"))))
