diff --git a/manifest/product_octez.ml b/manifest/product_octez.ml
index f4336a077a..6ab49b65cf 100644
--- a/manifest/product_octez.ml
+++ b/manifest/product_octez.ml
@@ -522,7 +522,7 @@ let octez_rust_deps =
               [S "source_tree"; S "../riscv"];
               [S "source_tree"; S "../kernel_sdk"];
             ];
-            [S "action"; [S "no-infer"; [S "bash"; S "./build.sh"]]];
+            [S "action"; [S "no-infer"; [S "system"; S "bash ./build.sh"]]];
           ];
         ]
 
diff --git a/src/rust_deps/dune b/src/rust_deps/dune
index de48dbfde8..20934e2e21 100644
--- a/src/rust_deps/dune
+++ b/src/rust_deps/dune
@@ -22,4 +22,4 @@
   (source_tree src)
   (source_tree ../riscv)
   (source_tree ../kernel_sdk))
- (action (no-infer (bash ./build.sh))))
+ (action (no-infer (system "bash ./build.sh"))))
