diff --git a/manifest/product_octez.ml b/manifest/product_octez.ml
index b99bf82156..3bcb1d9129 100644
--- a/manifest/product_octez.ml
+++ b/manifest/product_octez.ml
@@ -575,7 +575,7 @@ let octez_rust_deps =
               [S "file"; S "../../src/lib_wasm_runtime/build.rs"];
               [S "glob_files_rec"; S "../../src/lib_wasm_runtime/src/**.rs"];
             ];
-            [S "action"; [S "no-infer"; [S "bash"; S "./build.sh"]]];
+            [S "action"; [S "no-infer"; [S "system"; S "bash ./build.sh"]]];
           ];
         ]
 
diff --git a/src/rust_deps/dune b/src/rust_deps/dune
index 4e6bc21230..094f4f0d1e 100644
--- a/src/rust_deps/dune
+++ b/src/rust_deps/dune
@@ -25,4 +25,4 @@
   (file ../../src/lib_wasm_runtime/Cargo.toml)
   (file ../../src/lib_wasm_runtime/build.rs)
   (glob_files_rec ../../src/lib_wasm_runtime/src/**.rs))
- (action (no-infer (bash ./build.sh))))
+  (action (no-infer (system "bash ./build.sh"))))
