diff --git a/src/rust_deps/dune b/src/rust_deps/dune
index 94e7bb4f93..ce2fd1535b 100644
--- a/src/rust_deps/dune
+++ b/src/rust_deps/dune
@@ -25,4 +25,4 @@
   (file ../../src/lib_wasm_runtime/Cargo.toml)
   (file ../../src/lib_wasm_runtime/build.rs)
   (glob_files_rec ../../src/lib_wasm_runtime/src/**.rs))
- (action (no-infer (bash ./build.sh))))
+ (action (no-infer (system "bash ./build.sh"))))
diff --git a/manifest/product_octez.ml b/manifest/product_octez.ml
index 662c865151..2b6d02f824 100644
--- a/manifest/product_octez.ml
+++ b/manifest/product_octez.ml
@@ -525,7 +525,7 @@ let octez_rust_deps =
               [S "file"; S "../../src/lib_wasm_runtime/build.rs"];
               [S "glob_files_rec"; S "../../src/lib_wasm_runtime/src/**.rs"];
             ];
-            [S "action"; [S "no-infer"; [S "bash"; S "./build.sh"]]];
+            [S "action"; [S "no-infer"; [S "system"; S "bash ./build.sh"]]];
           ];
         ]
 
