diff --git a/dune b/dune
index adf795a359..5850db52d4 100644
--- a/dune
+++ b/dune
@@ -65,4 +65,4 @@
  (action
   (with-stdout-to
    %{target}
-   (bash etherlink/scripts/generate_node_link_flags.sh))))
+   (system "bash etherlink/scripts/generate_node_link_flags.sh"))))
diff --git a/etherlink/lib_wasm_runtime/dune b/etherlink/lib_wasm_runtime/dune
index 8b6d5d6d2a..34265c1552 100644
--- a/etherlink/lib_wasm_runtime/dune
+++ b/etherlink/lib_wasm_runtime/dune
@@ -23,4 +23,4 @@
   (source_tree ../../src/rust_deps/wasmer-3.3.0)
   (source_tree ../../src/kernel_sdk)
   (source_tree src))
- (action (no-infer (bash ./build.sh))))
+ (action (no-infer (system "bash ./build.sh"))))
diff --git a/manifest/product_etherlink.ml b/manifest/product_etherlink.ml
index 9d52eaa6b9..67ba7c4950 100644
--- a/manifest/product_etherlink.ml
+++ b/manifest/product_etherlink.ml
@@ -106,7 +106,7 @@ let evm_node_rust_deps =
               [S "source_tree"; S "../../src/kernel_sdk"];
               [S "source_tree"; S "src"];
             ];
-            [S "action"; [S "no-infer"; [S "bash"; S "./build.sh"]]];
+            [S "action"; [S "no-infer"; [S "system"; S "bash ./build.sh"]]];
           ];
         ]
 
diff --git a/manifest/product_octez.ml b/manifest/product_octez.ml
index f8e01986d8..a362f7ad9d 100644
--- a/manifest/product_octez.ml
+++ b/manifest/product_octez.ml
@@ -529,7 +529,7 @@ let octez_rustzcash_deps =
               [S "source_tree"; S "include"];
               [S "source_tree"; S "src"];
             ];
-            [S "action"; [S "no-infer"; [S "bash"; S "./build.sh"]]];
+            [S "action"; [S "no-infer"; [S "system"; S "bash ./build.sh"]]];
           ];
         ]
 
@@ -573,7 +573,7 @@ let octez_rust_deps =
               [S "source_tree"; S "../riscv"];
               [S "source_tree"; S "../kernel_sdk"];
             ];
-            [S "action"; [S "no-infer"; [S "bash"; S "./build.sh"]]];
+            [S "action"; [S "no-infer"; [S "system"; S "bash ./build.sh"]]];
           ];
         ]
 
diff --git a/src/rust_deps/dune b/src/rust_deps/dune
index 3da9488fcc..fc97e80d06 100644
--- a/src/rust_deps/dune
+++ b/src/rust_deps/dune
@@ -22,4 +22,4 @@
   (source_tree src)
   (source_tree ../riscv)
   (source_tree ../kernel_sdk))
- (action (no-infer (bash ./build.sh))))
+ (action (no-infer (system "bash ./build.sh"))))
diff --git a/src/rustzcash_deps/dune b/src/rustzcash_deps/dune
index 9c60b89c8e..1216e9782b 100644
--- a/src/rustzcash_deps/dune
+++ b/src/rustzcash_deps/dune
@@ -20,4 +20,4 @@
   (source_tree bin)
   (source_tree include)
   (source_tree src))
- (action (no-infer (bash ./build.sh))))
+ (action (no-infer (system "bash ./build.sh"))))
