diff --git a/etherlink/lib_wasm_runtime/dune b/etherlink/lib_wasm_runtime/dune
index e7f21cbaf1..2a4c583e68 100644
--- a/etherlink/lib_wasm_runtime/dune
+++ b/etherlink/lib_wasm_runtime/dune
@@ -25,4 +25,4 @@
   (source_tree ../../src/kernel_sdk)
   (source_tree ../../sdk/rust)
   (source_tree src))
- (action (no-infer (bash ./build.sh))))
+ (action (no-infer (system "bash ./build.sh"))))
diff --git a/manifest/product_etherlink.ml b/manifest/product_etherlink.ml
index fabeb54b10..d90de0dc26 100644
--- a/manifest/product_etherlink.ml
+++ b/manifest/product_etherlink.ml
@@ -110,7 +110,7 @@ let evm_node_rust_deps =
               [S "source_tree"; S "../../sdk/rust"];
               [S "source_tree"; S "src"];
             ];
-            [S "action"; [S "no-infer"; [S "bash"; S "./build.sh"]]];
+            [S "action"; [S "no-infer"; [S system "bash ./build.sh"]]];
           ];
         ]
 
diff --git a/manifest/product_octez.ml b/manifest/product_octez.ml
index faef74f977..29b6d2ff27 100644
--- a/manifest/product_octez.ml
+++ b/manifest/product_octez.ml
@@ -530,7 +530,7 @@ let octez_rustzcash_deps =
               [S "source_tree"; S "include"];
               [S "source_tree"; S "src"];
             ];
-            [S "action"; [S "no-infer"; [S "bash"; S "./build.sh"]]];
+            [S "action"; [S "no-infer"; [S "system"; S "bash ./build.sh"]]];
           ];
         ]
 
@@ -575,7 +575,7 @@ let octez_rust_deps =
               [S "source_tree"; S "../kernel_sdk"];
               [S "source_tree"; S "../../sdk/rust"];
             ];
-            [S "action"; [S "no-infer"; [S "bash"; S "./build.sh"]]];
+            [S "action"; [S "no-infer"; [S "system"; S "bash ./build.sh"]]];
           ];
         ]
 
diff --git a/src/rust_deps/dune b/src/rust_deps/dune
index 5ab1025177..18a0e9b509 100644
--- a/src/rust_deps/dune
+++ b/src/rust_deps/dune
@@ -23,4 +23,4 @@
   (source_tree ../riscv)
   (source_tree ../kernel_sdk)
   (source_tree ../../sdk/rust))
- (action (no-infer (bash ./build.sh))))
+ (action (no-infer (system "bash ./build.sh"))))
diff --git a/src/rustzcash_deps/dune b/src/rustzcash_deps/dune
index 9c60b89c8e..1216e9782b 100644
--- a/src/rustzcash_deps/dune
+++ b/src/rustzcash_deps/dune
@@ -20,4 +20,4 @@
   (source_tree bin)
   (source_tree include)
   (source_tree src))
- (action (no-infer (bash ./build.sh))))
+ (action (no-infer (system "bash ./build.sh"))))
